src/raftstore/store/store.rs

```diff
@@ -558,6 +558,37 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                     return Err(e);
                 }
             }
+
+            // If this peer detects the leader is missing for a long long time,
+            // it should consider itself as a stale peer which is removed from
+            // the original cluster.
+            // This most likely happens in the following scenario:
+            // At first, there are three node A, B, C in the cluster, and A is leader.
+            // Node B gets down. And then A adds D, E, F into the cluster.
+            // Node D becomes leader of the new cluster, and then removes node A, B, C.
+            // After all these node in and out, now the cluster has node D, E, F.
+            // If node B goes up at this moment, it still thinks it is one of the cluster
+            // and has peers A, C. However, it could not reach A, C since they are removed
+            // from the cluster or probably destroyed.
+            // Meantime, D, E, F would not reach B, since it's not in the cluster anymore.
+            // In this case, Node B would notice that the leader is missing for a long time,
+            // and it would check with pd to confirm whether it's still a member of the cluster.
+            // If not, it destroys itself as a stale peer which is removed out already.
+            if let Some(peer) = self.region_peers.get_mut(&region_id) {
+                let duration = peer.since_leader_missing();
+                if duration >= self.cfg.max_leader_missing_duration {
+                    info!("leader for peer {} missing for a long time, check with pd whether \
```

Comment:
Any better idea?
There is another case that the corresponding region info is not available in PD for some time.  So the PD worker should retry later (due to this message resent). And raftstore doesn't know how the worker is progressing.